packages/transmute-framework/test/EventStore/EventStore.spec.js

```diff
@@ -0,0 +1,236 @@
+
+  let fsa = getFSAFromEventArgs(event);
+
+  assert.equal(originalEvent.eventType, fsa.type);
+  assert.equal(originalEvent.key, Object.keys(fsa.payload)[0]);
+  assert.equal(originalEvent.value, fsa.payload[Object.keys(fsa.payload)[0]]);
+
+  return eventId;
+};
+
+const readEventFromAccount = async (
+  eventStore,
+  originalEvent,
+  eventId,
+  account
+) => {
+  let esEventValues = await eventStore.readEvent.call(eventId, {
+    from: account
+  });
+
+  assert.equal(esEventValues[0].toNumber(), eventId);
+  assert.equal(esEventValues[1], account);
+
+  fsa = getFSAFromEventValues(...esEventValues);
+  // console.log(fsa)
+  assert.equal(originalEvent.eventType, fsa.type);
+  assert.equal(originalEvent.key, Object.keys(fsa.payload)[0]);
+  assert.equal(originalEvent.value, fsa.payload[Object.keys(fsa.payload)[0]]);
+};
+
+describe("", () => {
+  let factory;
+
+  before(async () => {
+    factory = await EventStoreFactory.deployed();
+  });
+
+  contract("UnsafeEventStore", accounts => {
+    let eventStore;
+
+    it("the factory whitelist cannot be set by eventstore owner or creator", async () => {
+      factory
+        .setWhitelist(accounts, { from: accounts[3], gas: 2000000 })
+        .then(_tx => {
+          assert.fail(
+            "the factory whitelist can be set by eventstore owner or creator"
+          );
+        })
+        .catch(e => {
+          if (e.name == "Error") {
+            assert.ok(true);
+          } else {
+            assert.fail(
+              "the factory whitelist can be set by eventstore owner or creator"
+            );
+          }
+        });
+    });
+
+    it("the factory whitelist includes all web3 accounts", async () => {
+      await factory.setWhitelist(accounts, {
+        from: accounts[0],
+        gas: 2000000
+      });
+      let whitelist = await factory.getWhitelist({
+        from: accounts[0],
+        gas: 2000000
+      });
+      assert.deepEqual(whitelist, accounts);
+    });
+
+    it("the factory whitelist cannot be overwritten", async () => {
+      factory
+        .setWhitelist(_.slice(accounts, 0, 2), {
+          from: accounts[0],
+          gas: 2000000
+        })
+        .then(_tx => {
+          assert.fail("the factory whitelist can be overwritten");
+        })
+        .catch(e => {
+          if (e.name == "Error") {
+            assert.ok(true);
+          } else {
+            assert.fail("the factory whitelist can be overwritten");
+          }
+        });
+    });
+
+    it("the factory caller is the event store contract creator", async () => {
+      let _tx = await factory.createEventStore(_.slice(accounts, 0, 2), {
+        from: accounts[0],
+        gas: 2000000
+      });
+      let fsa = getFSAFromEventArgs(_tx.logs[0].args);
+      eventStore = EventStore.at(fsa.payload.address);
+      let creator = await eventStore.creator();
+      assert(creator === accounts[0]);
+    });
+
+    it("the factory is the event store owner", async () => {
```

Comment:
What happens when self destruct is called? Really wish there were code coverage for solidity. Every contract interface should be tested.

Reply:
Good point, will add this.

Reply:
Damn. I wonder how many deployed contracts do not protect against randoms selfdestructing

Reply:
yeah, we should make it a practice to cover that from now on. thanks for taking one for the team parity :)